ec/ecresource/projectresource/create_test.go

```diff
@@ -0,0 +1,302 @@
+package projectresource
+
+import (
+	"context"
+	"testing"
+
+	"github.com/elastic/terraform-provider-ec/ec/internal/gen/serverless/resource_elasticsearch_project"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	type testData struct {
+		api           api[resource_elasticsearch_project.ElasticsearchProjectModel]
+		modelHandler  modelHandler[resource_elasticsearch_project.ElasticsearchProjectModel]
+		req           resource.CreateRequest
+		expectedDiags diag.Diagnostics
+		expectedId    *string
+	}
+	tests := []struct {
+		name     string
+		testData func(context.Context) testData
+	}{
+		{
+			name: "should error if reading the tf model errors",
+			testData: func(ctx context.Context) testData {
+				req := resource.CreateRequest{
+					Plan: tfsdk.Plan{
+						Raw: tftypes.NewValue(tftypes.Bool, true),
+					},
+				}
+
+				api := NewMockapi[resource_elasticsearch_project.ElasticsearchProjectModel](ctrl)
+				api.EXPECT().Ready().Return(true)
+
+				readDiags := diag.Diagnostics{
+					diag.NewErrorDiagnostic("nope", "nope"),
+				}
+				handler := NewMockmodelHandler[resource_elasticsearch_project.ElasticsearchProjectModel](ctrl)
+				handler.EXPECT().ReadFrom(ctx, req.Plan).Return(nil, readDiags)
```

Comment:
If `req` is removed, it can become `handler.EXPECT().ReadFrom(ctx, tfsdk.Plan{}).Return(nil, readDiags)`